### DISCUSSION TITLE
Mapbox GL license change for new version

### DISCUSSION THREAD

**Discussion by thomasgwatson** [uber/react-map-gl#1380] [General]
Big news...

https://github.com/mapbox/mapbox-gl-js/blob/main/CHANGELOG.md#200

Curious to hear what the plan is from here. Could a different map render we substituted (Harp.gl, tangram, etc)?

**Comment by Pessimistress** [comment]
In the short term we are taking the following actions:

- Maintain parallel react-map-gl releases that depend on mapbox-gl v2 and v1 respectively, so that users are not forced to upgrade
- Update documentation regarding self-hosted tiles
- Provide documentation on how to substitute mapbox-gl dependency with a forked library (e.g. via Webpack module aliasing)

In the long term, yes we could consider supporting an alternative map renderer. InteractiveMap, Marker, Popup and NavigationControl work fairly independently from the underlying rendering engine. The other controls may not work unless we duplicate a significant amount of code from the mapbox-gl code base.

**Comment by testower** [comment]
So is there a release scheduled for the upgrade?

**Comment by Pessimistress** [comment]
v6.0.0 is published: https://github.com/visgl/react-map-gl/releases/tag/v6.0.0

**Comment by snickell** [comment]
@Pessimistress really appreciate your work on this!!!

**Comment by thomasgwatson** [comment]
For those curious about the "official" open source successor to MapboxGL, [MapLibre has been launched](https://github.com/maplibre/maplibre-gl-js). The repo can be found [here](https://github.com/maplibre/maplibre-gl-js).

I assume for the time being that integration with MapLibre will be required via the webpack module aliasing method suggested by @Pessimistress above.
